src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -135,11 +160,28 @@ var DraftEditorCompositionHandler = {
             EditorState.set(nextEditorState, {inCompositionMode: false}),
         );
     } else {
+      // If any children have been added/removed the reconciler will crash unless we restore the dom.
+      const mustReset = hasMutation;
+      if (mustReset) {
+        editor.restoreEditorDOM();
+      }
+
       const nextEditorState = replaceText(editor._latestEditorState, newText, compositionRange);
 
       editor.setMode('edit');
+      const editorStateProps = mustReset ? {
+        // TODO this is done in the draft composition handler, but I am not sure we should.
+        nativelyRenderedContent: null,
+        forceSelection: true,
+      } : {
+        // pass in nativelyRenderedContent here?
+        forceSelection: true, // Not sure this is needed…
```

Comment:
I don't think(?) this should be necessary? If there weren't any childList mutations, the selection should already be where it's supposed to go, at least in theory.

I think one of the problems I had one my other branch was that I was forcing selection every time. Since the rerender depends on a setState, it is asynchronous. When it rerenders, it'll also move the selection to wherever draft thinks the selection should be after the replaceText, even if the user made additional changes or moved the selection since.

Reply:
Yeah, since writing this I dug into draft a bit more and came to a similar conclusion. Let's merge this, then start making changes to allow native dom changes to go through.